astropy/io/misc/asdf/tags/transform/functional_models.py

```diff
@@ -0,0 +1,783 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+# -*- coding: utf-8 -*-
+
+from numpy.testing import assert_array_equal
+
+from asdf import yamlutil
+
+from astropy.modeling import functional_models
+from .basic import TransformType
+from . import _parameter_to_value
+
+
+__all__ = ['AiryDisk2DType', 'Box1DType', 'Box2DType', 
+           'Disk2DType', 'Ellipse2DType', 'Exponential1DType', 
+           'Gaussian1DType', 'Gaussian2DType', 'KingProjectedAnalytic1DType',
+           'Logarithmic1DType', 'Lorentz1DType', 'Moffat1DType',
+           'Moffat2DType', 'Planar2D', 'RedshiftScaleFactorType',
+           'RickerWavelet1DType', 'RickerWavelet2DType', 'Ring2DType',
+           'Sersic1DType', 'Sersic2DType', 'Sine1DType', 'Trapezoid1DType',
+           'TrapezoidDisk2DType', 'Voigt1DType']
+
+
+class AiryDisk2DType(TransformType):
+    name = 'transform/airy_disk2d'
+    version = '1.0.0'
+    types = ['astropy.modeling.functional_models.AiryDisk2D']
+
+    @classmethod
+    def from_tree_transform(cls, node, ctx):
+        return functional_models.AiryDisk2D(amplitude=node['amplitude'],
+                                            x_0=node['x_0'],
+                                            y_0=node['y_0'],
+                                            radius=node['radius'])
+
+    @classmethod
+    def to_tree_transform(cls, model, ctx):
+        node = {'amplitude': _parameter_to_value(model.amplitude),
+                'x_0': _parameter_to_value(model.x_0),
+                'y_0': _parameter_to_value(model.y_0),
+                'radius': _parameter_to_value(model.radius)}
+        return yamlutil.custom_tree_to_tagged_tree(node, ctx)
```

Comment:
Now that #10189 has been merged, this and other calls to `custom_tree_to_tagged_tree` and `tagged_tree_to_custom_tree` can be removed.